sympy/functions/elementary/hyperbolic.py

```diff
@@ -228,7 +228,16 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        if self.args[0].is_real:
+            return True
+
+    def _eval_is_positive(self):
```

Comment:
In your branch:
```python
>>> sinh(1 + I).is_positive
None

>>> sinh(1 + I).evalf().is_positive
False
```

Reply:
You probably want `sinh(1 + I).is_real` to be False instead of None (making `is_positive` False too), but I'm not sure it's worth it.

It would be easy to call `self.as_real_imag()` and then `.is_zero`, but even if we pass `deep=False` nested evaluation will lead to runtime exponential in the depth of the expression: 8 seconds for building `(sinh^9)(1 + I)` (`^` in the sense of nested application) and 24 seconds for `.is_real`, instead of 3 milliseconds total with the current PR. 

Slowing down to return True instead of None could have been worthwhile because proving something is real often leads to simplifications, but is it worth the slowdown to return False instead of None?

Food for thought: in the current implementation, `(a+b).is_real` only returns True when both a and b are real.

```py
>>> (exp(2*I*pi/3) + exp(4*I*pi/3)).is_real
None
```